### ISSUE TITLE
hoverPoints is wrong on column chart with shared tooltip

### ISSUE THREAD

**Issue by onemenny** [highcharts/highcharts#5914]
With shared tooltip the current hover point is wrong on column charts.
see here: http://jsfiddle.net/onemenny/5yem1b5n/ try to hover vertically from the bottom of the column to the next column value and than down again (yellow -> green, green -> yellow)

the point does not change when going down again, sometimes when going up. 
on the upper part (green:blue) it does work. 

This is a bug IMHO, 
How can I always get the current point under the mouse with shared tooltip, with all chart types?

**Comment by jon-a-nygaard** [comment]
Hi @onemenny 
The issue is has been fixed since v4.2.7. http://jsfiddle.net/5yem1b5n/6/

Internal: [Simplified demo](http://jsfiddle.net/5yem1b5n/5/)

**Comment by onemenny** [comment]
Ok, but 4.2.7 has other problem(s): https://github.com/highcharts/highcharts/issues/5899
I cannot upgrade because of that. Please provide a fix for one of these issues

**Comment by onemenny** [comment]
Please see github, I cannot upgrade to 4.2.7
I need a hot fix like you provided here:
https://github.com/highcharts/highcharts/issues/5839
(see my other support case)

**Comment by TorsteinHonsi** [comment]
@onemenny This issue was fixed by c83323a and the following d42b631, so you can try cherry-picking those commits into your old version.

**Comment by onemenny** [comment]
I just checked it with version 5.0.3 - the issue is NOT resolved yet. 
Just run it on column chart with stacking (normal). When you vertically move the mouse over 1 of the columns the hover point never changes. 
Moreever when you horizontally move between columns (drag move) - the hover point never changes as well. 

(I hope I get it wrong, cause I've waited for this fix for long time)

**Comment by jon-a-nygaard** [comment]
Hi @onemenny from what I can see this is fixed, am I missing something? [Here`s the example](http://jsfiddle.net/5yem1b5n/9/) which I recorded from below.
![](https://s15.postimg.org/w6vgo21a3/5914.gif)

**Comment by onemenny** [comment]
It does not work on mobile: touch drag

On Mon, Nov 21, 2016 at 10:06 AM, Jon Arild Nygård <notifications@github.com

**Comment by onemenny** [comment]
Hey, please reopen this issue, or answer please

**Comment by jon-a-nygaard** [comment]
Reopening due to chart hoverpoint is not updated on dragging in mobile browsers.
Tested with Chrome on Android.

**Comment by jon-a-nygaard** [comment]
I tried to create a workaround for this issue by not using shared tooltip, but I found that it then checks wether [hoverPoint is different from previous hoverPoint](https://github.com/highcharts/highcharts/blob/master/js/parts/Pointer.js#L243). And hoverPoint is updated in onMouseOver, but onMouseOver is only called when [the previous mentioned condition is true](https://github.com/highcharts/highcharts/blob/master/js/parts/Pointer.js#L262).
In other words, hoverPoint will almost never be updated, at least when it comes to touch events.

I am considering moving the KDPoints logic to its own function, to seperate concerns a bit more. 
@pawelfus Torstein told me that you have some experience with the `Pointer.runPointActions`, do you perhaps have some suggestion and insight to this?

**Comment by pawelfus** [comment]
I'm not sure about the top part - it's a general problem or touch devices related only? Is this the example you mentioned: http://jsfiddle.net/5yem1b5n/22/ ?

Regarding the original issue, if this works for mouse events, but not for touch, then it's strange - `runPointActions()` is exactly the same in both cases, right?

As for this:
> I am considering moving the KDPoints logic to its own function, to seperate concerns a bit more.

+1 !

> (...) do you perhaps have some suggestion and insight to this?

Yes, a lot of troubles I had with [`byProximity`](https://github.com/highcharts/highcharts/blob/master/js/parts/Interaction.js#L483). Maybe that's another case?

**Comment by onemenny** [comment]
Hello @jon-a-nygaard @TorsteinHonsi 
What is the status of this bug? It has been nearly 1.5 month since the case was opened. 
Column chart tooltip is unusable, Our customers cannot use it to get the right value. This is extremely CRITICAL for us 

Please help.

**Comment by jon-a-nygaard** [comment]
Hi @onemenny , this bug is still in development. However you can try out the latest progress of [http://github.highcharts.com/dev-runPointActions/highcharts.js](http://github.highcharts.com/dev-runPointActions/highcharts.js). It might work better for your case, altough it is not considered stable.

Here's an [example](http://jsfiddle.net/jon_a_nygaard/5yem1b5n/27/) of a possible workaround. Let me know if it works.

**Comment by onemenny** [comment]
@jon-a-nygaard 
the hover point does not change properly - only at the middle of the stacked column it changes. 
on tooltip formatter, even when this.points has actual value hoverPoint may be null - I do not comprehend this behaviour 
mobile device seems to be ok but I need to do more testing.

**Comment by jon-a-nygaard** [comment]
Fixed as of version 5.0.8, but advisable to skip to 5.0.9 due to an error with 3D-columns in the compiled file of `highcharts-3d.js`.

**Comment by onemenny** [comment]
@jon-a-nygaard  Indeed this is fixed in version 5.0.8 and 5.0.9 but in 5.0.10 this does not work 
You can use my fiddle at the top of the thread, in the middle of the column point, hover point changes instead of when leaving the point area 

Please reopen this issue

**Comment by jon-a-nygaard** [comment]
Hi @onemenny , this is due to a regression which is fixed in the next version. You can see the details of the fix in issue #6586.
